src/read/xcoff/symbol.rs

```diff
@@ -251,9 +278,22 @@ impl<'data, 'file, Xcoff: FileHeader, R: ReadRef<'data>> ObjectSymbol<'data>
 
     #[inline]
     fn size(&self) -> u64 {
-        // TODO: get the symbol size when the csect auxiliary symbol is supported.
-        // Most symbols don't have sizes.
-        0
+        if self.symbol.has_aux_csect() {
+            // XCOFF32 must have the csect auxiliary entry as the last auxiliary entry.
+            // XCOFF64 doesn't require this, but conventionally does.
```

Comment:
If it's not required, then ideally we would handle it. If we don't, we must at least validate that `x_auxtype == _AUX_CSECT`.

Reply:
Added these validations in `fn aux_file()` and `fn aux_csect`.